Implemented cec.close()

jonathancyu opened the pull request:
I found that killing the process or finishing the script locks out this library from further use in other scripts, as well as the `cec-utils` command line utility.

jonathancyu commented:
I would squash commits btw :)

trainman419 commented on cec.cpp:
```diff
@@ -661,6 +677,7 @@ PyObject * persist_config(PyObject * self, PyObject * args) {
 static PyMethodDef CecMethods[] = {
    {"list_adapters", list_adapters, METH_VARARGS, "List available adapters"},
    {"init", init, METH_VARARGS, "Open an adapter"},
+   {"close", close, METH_VARARGS, "Close an adapter"},
```
Since this function takes no arguments, this should be METH_NOARGS.

trainman419 commented on cec.cpp:
```diff
@@ -259,6 +262,19 @@ static PyObject * init(PyObject * self, PyObject * args) {
 
+static PyObject * close(PyObject * self, PyObject * args) {
+
+   if( PyArg_ParseTuple(args, "") == 0 && CEC_adapter == NULL ) {
```
Since this function takes no arguments, you don't need to parse `args`.

I think the condition here should be `CEC_adapter != NULL`

jonathancyu replied:
yep you're right, whoops

trainman419 commented:
Thanks! Would you mind updating the readme as well?

jonathancyu commented:
I changed the description on the readme, added cec.close() to cb_test.py, and also took out destroy_adapter - I don't think destroying the adapter is desired, and calling functions after close() will segfault. 

One thing to note is that 
```cec.init()
cec.close()
cec.init()```
will hang on the last init(), so this'll break code that closes and opens the adapters.

trainman419 commented:
> will hang on the last init(), so this'll break code that closes and opens the adapters.

It wasn't even possible to close and reopen the adapter before, so I don't think this is a big issue.

Thank you for implementing this!
